### ISSUE TITLE
Athena: Table aliases with columns lead to unsolvable L025

### ISSUE THREAD

**Issue by sbrandtb** [sqlfluff/sqlfluff#3677]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using table aliases with column lists as *only* table source, L025 alias is never used in select statement cannot be fixed (At least I could not.)

### Expected Behaviour

No L025 or error raised in all of the examples.

`unnest2.sql` is only here to show that the usual way of marking something unused (at least in Python, by convention) by naming it `_` does not work and even raise a parser error although it is valid code. I will open a separate issue for it, this one here is simply about L025!

Documentation on syntax is as always sparse in PrestoDB/Athena, but [here](https://prestodb.io/docs/current/sql/select.html#unnest) the relevant part on subtables with column lists.

### Observed Behaviour

L025 shown.

```
== [unnest1.sql] FAIL                                                                                                                                                                                                
L:   4 | P:   6 | L025 | Alias 'foo' is never used in SELECT statement.
```

```
== [unnest2.sql] FAIL                                                                                                                                                                                                
L:   4 | P:   3 |  PRS | Line 4, Position 3: Found unparsable section: 'AS _(a)'
```

### How to reproduce

```sql
-- unnest1.sql
SELECT a
FROM (
    VALUES ('a'), ('b')
) AS foo(a)

-- unnest2.sql
SELECT a
FROM (
    VALUES ('a'), ('b')
) AS _(a)
```

### Dialect

Athena

### Version

sqlfluff --version
sqlfluff, version 1.2.1

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
IIRC, one or more of the dialects support parsing **CTE** names with a parenthesized column list. I wonder if we can reuse some of that work (i.e. grammar or segment definitions).

**Comment by keraion** [comment]
Was fixed by #5726
